src/omnisharpDownload.ts

```diff
@@ -0,0 +1,120 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+ 
+'use strict';
+
+import * as fs from 'fs-extra-promise';
+import * as path from 'path';
+import * as tmp from 'tmp';
+
+const Decompress = require('decompress');
+const Github = require('github-releases');
+
+const OmnisharpRepo = 'OmniSharp/omnisharp-roslyn';
+const OmnisharpVersion = 'v1.9-alpha3';
+const DefaultInstallLocation = path.join(__dirname, '../.omnisharp');
+
+tmp.setGracefulCleanup();
+
+function getOmnisharpAssetName(): string {
+    switch (process.platform) {
+        case 'win32':
+            return 'omnisharp-win-x64-dnx451.zip';
```

Comment:
One question - I noticed that currently these zip files contain a full .NET Framework. Do we want to do something before RC2 so that we share these things?